src/components/network/NetworkItem.vue

```diff
@@ -5,7 +5,7 @@
   >
     <div class="network-icon">
       <img
-        :src="`img/networks/${network.id}.png`"
+        :src="`${network.icon}`"
```

Comment:
nice

Reply:
where is the icon comming from?

Reply:
nice, but remember to do smaller PRs and don't do everything in one PR. this now requires the API to be deployed before this PR

Reply:
The API PR for this was really concise and has already been merged! :wink: 

You are right this PR has perhaps grown too big, but we needed it like this in order to have the submodule to work on the extension concurrently.